src/Node.java

```diff
@@ -0,0 +1,42 @@
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+
+public class Node {
```

Comment:
There's an oportunity to reduce the contract area even further. This class should not be public really. It's not part of the API of the class after all, i.e. it's really an something internal to the trie - it exists solely to simplify the implementation.
It's small and simple enough to even move it within the PrefixTree file.